libraries/AP_HAL_PX4/RCOutput.cpp

```diff
@@ -495,8 +469,7 @@ void PX4RCOutput::_send_outputs(void)
                             ap_uc->rco_write(_period[i], i);
                         }
 
-                        bool armed = hal.util->get_soft_armed();
-                        if (armed) {
+                        if (hal.util->safety_switch_state() != AP_HAL::Util::SAFETY_DISARMED) {
```

Comment:
I would prefer `hal.util->get_soft_armed()` to stay as it is correct way to check for arming. Can be moved into `if()` though.